pkg/ebpf/probes/probes.go

```diff
@@ -124,16 +124,6 @@ func Init(module *bpf.Module, netEnabled bool) (Probes, error) {
 		DoMmapRet:                  &traceProbe{eventName: "do_mmap", probeType: kretprobe, programName: "trace_ret_do_mmap"},
 	}
 
-	// disable autoload for tcProbes if network is disabled
```

Comment:
We should disable autoload for the new cgroup programs or otherwise CAP_NET_ADMIN will be required even if we don't choose network events.

Reply:
My idea was to get rid of specifics and make things generic enough so that network probes or regular tracing probes were configured the same, specially after the introduction of privilege rings (which even having CAP_NET_ADMIN, that capability won't be effective after initialization either way). 

I guess I can keep the same "if netEnabled() logic disable autoload" and then enable them at tracee probes.Init().

Reply:
Okay so CAP_NET_ADMIN is only enabled if user is tracing net events OR capturing network:

```
rafaeldtinoco@fujitsu:~/.../tracee-mycode$ sudo ./dist/tracee-ebpf --log debug --install-path /home/rafaeldtinoco/tracee/tmp --cache cache-type=mem --cache mem-cache-size=512 --output format:json --output option:parse-arguments --output option:detect-syscall --trace comm=ping --capture net --capabilities bypass=false 2>&1 | grep net_admin
{"level":"debug","ts":1673872318.1034772,"msg":"enabling cap","cap":"cap_net_admin","pkg":"capabilities","file":"capabilities.go","line":313}
```
```
rafaeldtinoco@fujitsu:~/.../tracee-mycode$ sudo ./dist/tracee-ebpf --log debug --install-path /home/rafaeldtinoco/tracee/tmp --cache cache-type=mem --cache mem-cache-size=512 --output format:json --output option:parse-arguments --output option:detect-syscall --trace comm=ping --trace event=net_packet_ipv4 --capabilities bypass=false 2>&1 | grep net_admin
{"level":"debug","ts":1673872340.8947747,"msg":"enabling cap","cap":"cap_net_admin","pkg":"capabilities","file":"capabilities.go","line":313}
```
```
rafaeldtinoco@fujitsu:~/.../tracee-mycode$ sudo ./dist/tracee-ebpf --log debug --install-path /home/rafaeldtinoco/tracee/tmp --cache cache-type=mem --cache mem-cache-size=512 --output format:json --output option:parse-arguments --output option:detect-syscall --trace comm=ping --capabilities bypass=false 2>&1 | grep net_admin
```

Reply:
> `rafaeldtinoco@fujitsu:~/.../tracee-mycode$ sudo ./dist/tracee-ebpf --log debug --install-path /home/rafaeldtinoco/tracee/tmp --cache cache-type=mem --cache mem-cache-size=512 --output format:json --output option:parse-arguments --output option:detect-syscall --trace comm=ping --capabilities bypass=false 2>&1 | grep net_admin`

I don't think this says that CAP_NET_ADMIN is not used to load the BPF programs, but just that it is not requested.
Try to run tracee without CAP_NET_ADMIN privilege and see if you get an error (I think it will fail while trying to load the cgroup programs)

Reply:
Ok, I see that you reintroduced it below, so that should be ok